### DISCUSSION TITLE
Run Build Against Cypress Tests Locally

### DISCUSSION THREAD

**Discussion by followbl** [vercel/next.js#14351] [Help] [answered]
Currently I don't believe there's a way to run a `serverless` build of a next.js application locally against a suite of cypress tests. (or to simplify - just simply run a `serverless` build of a next.js application locally.

We're trying to run an e2e cypress test suite against a build in our ci (github-actions). The only way we're having any success is to run it against `next dev`. Would love to have the tests run more efficiently if it was running against a build.

note: Only reason we're not running it against an actual vercel build-preview is we'd like to have full control over the services it's interacting with: i.e. spinning up database & hasura via docker-compose.

**Comment by timneutkens** [comment, marked as answer]
`next start` can boot up a serverless build, however that should not be used in production as the serverless output is not optimized for a single-server case.

**Reply by followbl** [reply]
@timneutkens thanks for this - took a couple days, we were scared off by knex and knex drivers that worked in our production deployed build - but did not locally when we attempted to use the serverless locally (on the ci)...anyways your answer gave us confidence to keep going - so thank you 🎉
